modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -90,29 +103,54 @@ public Object execute(OperationExecutor executor, OperationContext context) thro
         }
     }
 
-    private MutableConfigurationStats getMutableConfigurationStats(OperationContext context)
+    private Object executeWithRetryPolicy(OperationExecutor executor, OperationContext context, List<Interceptor> interceptors) throws Throwable
     {
-        ConfigurationStats stats = context.getConfiguration().getStatistics();
-        return stats instanceof MutableConfigurationStats
-               ? (MutableConfigurationStats) stats
-               : null;
+        final RetryPolicyTemplate retryPolicyTemplate;
+        final Optional<ConnectionProvider> connectionProvider;
+
+        connectionProvider = context.getConfiguration().getConnectionProvider();
+
+        retryPolicyTemplate = getRetryPolicyTemplateFromConnectionProvider(connectionProvider);
+        LifecycleUtils.initialiseIfNeeded(retryPolicyTemplate, muleContext);
```

Comment:
yes, just one per each element in the XML. Doing further digging into your code, the whole thing doesn't seem necessary.

You either take the retry policy from the connection provider (in which case the LifecycleAwareConnectionProvider should have properly initialised the retry policy on its own initialise() phase), or you get a default from the ConnectionManagerAdapter (which should also have initialised it). In either case, you shouldn't do anything here